modules/navigation/src/main/java/com/sygic/maps/module/navigation/NavigationFragment.kt

```diff
@@ -29,35 +29,104 @@ import android.util.AttributeSet
 internal const val KEY_PREVIEW_MODE = "preview_mode"
 internal const val KEY_ROUTE_INFO = "route_info"
 
 /**
  * A *[NavigationFragment]* is the core component for any navigation operation. It can be easily used for the navigation
- * purposes. By setting the [routeInfo] object will start the navigation process. Any pre build-in element such as [DirectionInfo]
- * [Signposts], [Infobar], [CurrentSpeed] or [SpeedLimit] may be activated or deactivated and styled.
+ * purposes. By setting the [routeInfo] object will start the navigation process. Any pre build-in element such as
+ * [FullSignpostView], [SimplifiedSignpostView], [Infobar], [CurrentSpeed] or [SpeedLimit] may be activated or deactivated and styled.
  */
 @Suppress("unused", "MemberVisibilityCanBePrivate")
 class NavigationFragment : MapFragmentWrapper<NavigationFragmentViewModel>() {
 
     override lateinit var fragmentViewModel: NavigationFragmentViewModel
 
+    @Inject
+    internal lateinit var regionalManager: RegionalManager
+
     override fun executeInjector() =
         injector<NavigationComponent, NavigationComponent.Builder>(DaggerNavigationComponent.builder()) { it.inject(this) }
 
+    /**
+     * A *[distanceUnit]* defines all available [DistanceUnit]'s type.
+     *
+     * [DistanceUnit.KILOMETERS] (default) -> Kilometers/meters are used as the distance unit.
+     *
+     * [DistanceUnit.MILES_YARDS] -> Miles/yards are used as the distance unit.
+     *
+     * [DistanceUnit.MILES_FEETS] -> Miles/feets are used as the distance unit.
+     */
+    var distanceUnit: DistanceUnit
+        get() = if (::fragmentViewModel.isInitialized) {
+            regionalManager.distanceUnit.value!!
+        } else arguments.getParcelableValue(KEY_DISTANCE_UNITS) ?: DISTANCE_UNITS_DEFAULT_VALUE
+        set(value) {
+            arguments = Bundle(arguments).apply { putParcelable(KEY_DISTANCE_UNITS, value) }
+            if (::fragmentViewModel.isInitialized) {
+                regionalManager.distanceUnit.value = value
+            }
+        }
+
+    /**
+     * A *[signpostEnabled]* modifies the SignpostView ([FullSignpostView] or [SimplifiedSignpostView]) visibility.
+     * Note, this need to be set before the [NavigationFragment] commit transaction.
+     *
+     * @param [Boolean] true to enable the SignpostView, false otherwise.
+     *
+     * @return whether the SignpostView is on or off.
+     */
+    var signpostEnabled: Boolean
+        get() = if (::fragmentViewModel.isInitialized) {
+            fragmentViewModel.signpostEnabled
+        } else arguments.getBoolean(KEY_SIGNPOST_ENABLED, SIGNPOST_ENABLED_DEFAULT_VALUE)
+        set(value) {
+            arguments = Bundle(arguments).apply { putBoolean(KEY_SIGNPOST_ENABLED, value) }
+        }
+
+    /**
+     * A *[signpostType]* defines the SignpostView ([FullSignpostView] or [SimplifiedSignpostView]) type to be used.
+     * Note, this need to be set before the [NavigationFragment] commit transaction.
```

Comment:
rozmyslam, ci by sme nenechali toto len na xml atribut. Lebo pocas behu to nemenime a tym padom je ten setter taky zavadzajuci, aj ked v dokumentacii to pises... Je to nieco, co musi byt zname pri konstruovani, takze by to malo byt v "konstruktore", alebo v xml. A v xml to mame a konstruktor by som neriesil. Ci?